src/RepositoryProxy.php

```diff
@@ -101,7 +101,15 @@ public function assertNotExists($criteria, string $message = ''): self
      */
     public function first(): ?Proxy
     {
-        return $this->findOneBy([]);
+        return $this->findOneBy([], ['id' => 'ASC']);
```

Comment:
```suggestion
        return $this->findBy([], [$sort => 'ASC'], 1)[0] ?? null;
```

The second parameter of `findOneBy` isn't necessarily available as it is not defined on the `ObjectRepository` interface.

Reply:
The project isn't limited to the EntityManager ? I ask it because this interface work with the Entity and Document Manager I think that's the reason about the missmatch.

But if we really want to deal with the interface (not only the EntityRepository), I think we must create an other `findOneBy` with an other name (that don't exist in the interface) to be sure to not break it.
Actually, if someone do not use the EntityManager, the `findOneBy` will be broken since the previous merge (it would try to add $orderBy in the arguments of the not EntityManager and... crash).

Reply:
Yes, this lib technically only requires `doctrine/persistence` which provides `ObjectManager::findOneBy(array $criteria)` (no `$orderBy`). It "should" be compatible with any manager/repository that implements this interface (ORM/ODM/etc...).

I have yet to see any user use something other than the ORM but I'd like to try and keep compatible as much as possible.

The ORM's `EntityRepository` adds `$orderBy` as an optional parameter. The MongoDB-ODM's `DocumentRepository` does not.

You're right, the `$orderBy` will potentially not be available with every `ObjectRepository` instance but I don't think it will crash, it will just ignore the extra parameter. This isn't great either. Probably should use reflection to see if this parameter exists - I've created an issue (#65).

For this PR, I'd like to use the method I described to make compatible with all instances of `ObjectRepository`'s.

Reply:
Yep, I've done that.

https://github.com/doctrine/orm/blob/2.7/lib/Doctrine/ORM/EntityRepository.php#L189

https://github.com/doctrine/mongodb-odm/blob/master/lib/Doctrine/ODM/MongoDB/Repository/DocumentRepository.php#L166
I think that it will have a problem if we pass it an extra argument.

Reply:
Yeah, it won't raise an exception but it will silently ignore the extra passed parameter (https://3v4l.org/RR1YS)